src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMSource.java

```diff
@@ -440,13 +440,15 @@ protected final void retrieve(@CheckForNull SCMSourceCriteria criteria,
             String fullName = repoOwner + "/" + repository;
             final GHRepository repo = github.getRepository(fullName);
             listener.getLogger().format("Looking up %s%n", HyperlinkNote.encodeTo(repo.getHtmlUrl().toString(), fullName));
-            try {
-                repositoryUrl = repo.getHtmlUrl();
-                collaboratorNames = new HashSet<>(repo.getCollaboratorNames());
-            } catch (FileNotFoundException e) {
-                // not permitted
-                listener.getLogger().println("Not permitted to query list of collaborators, assuming none");
-                collaboratorNames = Collections.emptySet();
+            if (!repo.isPrivate()) { // random people cannot create pull requests to private repositories
```

Comment:
That is worse. Seriously, all the distraction here is just slowing me down from actually fixing the real issue. Reducing API usage will not fix the problem, because you'll get more branches and eventually end up over the limit anyway. 

The fix is to throttle the API usage so that the operation takes longer but completes successfully.

After that fix lands (hopefully next week... but you know people keep distracting me 😉 ), then I can focus on improving the configuration UI, at which point changes like this will be easier to roll in without death by a hundred check boxes. (And death by a hundred checkboxes is infinitely preferable to death by ten system properties)